scheduler/src/cook/components.clj

```diff
@@ -61,15 +62,28 @@
                 [:headers "Cache-control"]
                 "max-age=0"))))
 
+(def rate-limiters
+  {:create-job-rate-limiter
+   (fnk [[:settings [:rate-limit expire-minutes job-creation]]]
+     (if job-creation
+       (let [{:keys [bucket-size is-enforcing rate-per-minute]} job-creation]
+         (when-not (or (true? is-enforcing) (false? is-enforcing))
+           (throw (ex-info ":is-enforcing not a boolean " job-creation)))
+         ((util/lazy-load-var 'ratelimit/tbf-ratelimiter-make) bucket-size rate-per-minute expire-minutes is-enforcing))
+       ; No config: Ratelimit engine is NOOP.
+       ratelimit/NOOP-RateLimiter))})
+
 (def raw-scheduler-routes
-  {:scheduler (fnk [mesos mesos-leadership-atom pool->pending-jobs-atom framework-id settings]
+  {:rate-limiters rate-limiters
+   :scheduler (fnk [mesos mesos-leadership-atom pool->pending-jobs-atom framework-id [:rate-limiters create-job-rate-limiter] settings]
                 ((util/lazy-load-var 'cook.mesos.api/main-handler)
                   datomic/conn
                   framework-id
                   (fn [] @pool->pending-jobs-atom)
                   settings
                   (get-in mesos [:mesos-scheduler :leader-selector])
-                  mesos-leadership-atom))
+                  mesos-leadership-atom
+                  create-job-rate-limiter))
```

Comment:
I think we should consider initializing the rate limiter object in `config.clj` and avoid having to thread it through to the `create-jobs!` handler.